test_cli.py

```diff
@@ -1113,16 +1118,9 @@ def test(ctx):
         ]
     )
     wait(remote_uuid)
-    # Since shared file system workers don't upload, exclude_patterns do not apply.
```

Comment:
@nelson-liu This test is always failing in GitHub Actions. It seems like `exclude_patterns` gets applied irregardless of whether a shared file system is used or not. Is this an acceptable behavior?

Reply:
Oops, yes that is the intended behavior—the test is incorrect. I was initially trying to make the sharedfs case special, but decided against it (so, `exclude_patterns` are applied in all cases). I forgot to update the test, but things worked anyway because checking `"CODALAB_SHARED_FILE_SYSTEM"` actually doesn't do anything on travis (i.e., it always evaluates to false). Here's a diff i'd apply, sorry for the confusion!

```diff
diff --git a/test_cli.py b/test_cli.py
index 60a18ac8..7d3da5d3 100644
--- a/test_cli.py
+++ b/test_cli.py
@@ -1113,16 +1113,9 @@ def test(ctx):
         ]
     )
     wait(remote_uuid)
-    # Since shared file system workers don't upload, exclude_patterns do not apply.
-    # Verify that all files are kept if the worker is using a shared file system.
-    if os.environ.get("CODALAB_SHARED_FILE_SYSTEM"):
-        check_num_lines(
-            2 + 2 + 3, _run_command([cl, 'cat', remote_uuid])
-        )  # 2 header lines, 1 stdout file, 1 stderr file, 3 items at bundle target root
-    else:
-        check_num_lines(
-            2 + 2 + 1, _run_command([cl, 'cat', remote_uuid])
-        )  # 2 header lines, 1 stdout file, 1 stderr file, 1 item at bundle target root
+    check_num_lines(
+        2 + 2 + 1, _run_command([cl, 'cat', remote_uuid])
+    )  # 2 header lines, 1 stdout file, 1 stderr file, 1 item at bundle target root

     # Test multiple exclude_patterns
     remote_uuid = _run_command(
@@ -1136,16 +1129,9 @@ def test(ctx):
         ]
     )
     wait(remote_uuid)
-    # Since shared file system workers don't upload, exclude_patterns do not apply.
-    # Verify that all files are kept if the worker is using a shared file system.
-    if os.environ.get("CODALAB_SHARED_FILE_SYSTEM"):
-        check_num_lines(
-            2 + 2 + 3, _run_command([cl, 'cat', remote_uuid])
-        )  # 2 header lines, 1 stdout file, 1 stderr file, 3 items at bundle target root
-    else:
-        check_num_lines(
-            2 + 2 + 1, _run_command([cl, 'cat', remote_uuid])
-        )  # 2 header lines, 1 stdout file, 1 stderr file, 1 item at bundle target root
+    check_num_lines(
+        2 + 2 + 1, _run_command([cl, 'cat', remote_uuid])
+    )  # 2 header lines, 1 stdout file, 1 stderr file, 1 item at bundle target root


 @TestModule.register('read')
```

Reply:
@nelson-liu No problem!